providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -441,7 +441,7 @@ static void *gen_init(void *provctx, int selection, int rsa_type,
             gctx->rsa_type = rsa_type;
         }
     }
-    if (!rsa_gen_set_params(gctx, params)) {
+    if (gctx != NULL && !rsa_gen_set_params(gctx, params)) {
```

Comment:
Why not just move it inside the if and replace the `gctx = NULL` above with `return NULL`

Reply:
Done and refactored a little.